src/TrackerCouncil.Smz3.Data/WorldData/Regions/IHasLocations.cs

```diff
@@ -21,4 +21,25 @@ public interface IHasLocations
     /// Gets all locations in the area.
     /// </summary>
     IEnumerable<Location> Locations { get; }
+
+    public IHasTreasure? GetTreasureRegion()
+    {
+        return Region as IHasTreasure;
+    }
+
+    public Region? Region => this switch
+    {
+        Room room => room.Region,
+        Region region => region,
+        _ => null
+    };
+
+    public bool IsKeysanityForArea
```

Comment:
I feel like the "ForArea" in the name here just adds confusion. Wouldn't just "IsKeysanity" be enough?

Reply:
So TrackerItemService and TrackerLocationService are for tracking things, and the ItemService and WorldService are more for querying data from the world(s), so to speak.

I do think there's a bit of shuffling things around a bit, though. The ItemService also weirdly enough also has stuff to get Rewards and Bosses. We could move all of that stuff into the WorldService, and if we want to rename it to be like "WorldQueryService" or something, I think that would make sense. It's for looking up locations, items, rewards, and bosses in the world (or worlds if playing a multiworld).

That would leave the main important thing in the ItemService the creation and caching of the two progression states. So maybe that could be renamed "ProgressionService"?